Code/Editor/Plugins/EditorAssetImporter/AssetImporterWindow.cpp

```diff
@@ -651,6 +660,53 @@ void AssetImporterWindow::HandleAssetLoadingCompleted()
     //  show the main area where all the actual work takes place
     ui->m_initialBrowseContainer->hide();
     m_rootDisplay->show();
+
+   
+    m_qtFileWatcher.addPath(m_fullSourcePath.c_str());
+    m_qtFileWatcher.addPath(m_assetImporterDocument->GetScene()->GetManifestFilename().c_str());
+}
+
+void AssetImporterWindow::FileChanged(QString path)
+{
+    if (m_isSaving)
+    {
+        return;
+    }
+
+    QString promptMessage(tr("The file %1 has been changed outside of the scene settings tool. This tool will be reloaded."));
+
+    // There currently isn't a way to cache unsaved changes, and attempt to re-apply them after reloading.
+    if (m_rootDisplay->HasUnsavedChanges())
+    {
+        promptMessage = tr("The file %1 has been changed outside of the scene settings tool. This tool will be reloaded and any unsaved changes will be lost. \n\n"
+                           "To prevent this from occuring in the future, do not modify the scene file or scene manifest outside of this tool while this tool has unsaved work.");
+    }
```

Comment:
you can probably do this in a closure

QString promptMessage = [&m_rootDisplay]() {
   if(m_rootDisplay->HasUnsavedChanges()) 
  {
    return tr("The file %1 has been changed outside of the scene settings tool. This tool will be reloaded and any unsaved changes will be lost. \n\n"
                           "To prevent this from occuring in the future, do not modify the scene file or scene manifest outside of this tool while this tool has unsaved work.");
  }
return  tr("The file %1 has been changed outside of the scene settings tool. This tool will be reloaded.");
}();